ddtrace/profiling/collector/_task.pyx

```diff
@@ -49,19 +50,45 @@ else:
     _gevent_tracer = DDGreenletTracer()
 
 
+cdef _asyncio_task_get_frame(task):
+    coro = task._coro
+    if hasattr(coro, "cr_frame"):
+        # async def
+        return coro.cr_frame
+    elif hasattr(coro, "gi_frame"):
+        # legacy coroutines
+        return coro.gi_frame
+    elif hasattr(coro, "ag_frame"):
+        # async generators
+        return coro.ag_frame
+    # unknown
+    return None
+
+
 cpdef get_task(thread_id):
     """Return the task id and name for a thread."""
+    task_id = None
+    task_name = None
+    frame = None
+
+    policy = _asyncio.get_event_loop_policy()
+    if isinstance(policy, _asyncio.DdtraceProfilerEventLoopPolicy):
+        loop = policy._ddtrace_get_loop(thread_id)
+        if loop is not None:
+            task = _asyncio.asyncio.tasks.current_task(loop)
+            if task is not None:
+                task_id = id(task)
+                task_name = task.get_name()
+                frame = _asyncio_task_get_frame(task)
+
     # gevent greenlet support:
-    # we only support tracing tasks in the greenlets are run in the MainThread.
-    if thread_id == nogevent.main_thread_id and _gevent_tracer is not None:
+    # - we only support tracing tasks in the greenlets are run in the MainThread.
+    # - if both gevent and asyncio are in use (!) we only return asyncio
```

Comment:
Should we note both of these in the public docs too? corp docs or at least our API docs (which doesn't really have a section for profiling so probably corp docs is best?)

Reply:
I'm not sure putting this kind of detail in the corp doc is a good idea.
I'd rather have it in the project doc if we have to document that limitation.